File: pyjl/tests/tests_new/if_else_test.py
def something():
    return "test"

def lookup_and_write(values):
    output = None
    if len(values) == 1:
         output = values[0]
    elif len(values) == 2:
         output = values[1]
    elif len(values) == 3:
         output = values[2]
    else:
        output = values
        
    return output

def lookup_and_write_without_else(values):
    output = None
    if len(values) == 1:
         output = values[0]
    elif len(values) == 2:
         output = values[1]
    elif len(values) == 3:
         output = values[2]
    return output

if __name__ == "__main__":
    assert lookup_and_write([]) == []
    assert lookup_and_write([1]) == 1
    assert lookup_and_write([1,2]) == 2
    assert lookup_and_write([1,2,3]) == 3
    assert lookup_and_write([1,2,3,4]) == [1,2,3,4]

    assert lookup_and_write_without_else([]) == None
    assert lookup_and_write_without_else([1]) == 1
    assert lookup_and_write_without_else([1,2]) == 2
    assert lookup_and_write_without_else([1,2,3]) == 3
    assert lookup_and_write_without_else([1,2,3,4]) == None

    print("OK")

File: pyjl/tests/tests_new-py2many/if_else_test.jl
function something()
println("Something");
end

function lookup_and_write(values, start, stop)
if isinstance(values, bytearray)
output = values
elseif 1 == 2
println("Never reach");
elseif 2 == 1
println("Never reach");
elseif 3 == 1
println("Never reach");
else
output = bytearray()
output[(begin + 1):stop - start] = something()
end

function main()
lookup_and_write([1, 2], 0, 1);
end

main()
